kafka-rest/src/test/java/io/confluent/kafkarest/v2/KafkaConsumerManagerTest.java

```diff
@@ -75,7 +78,7 @@
     private static final String topicName = "testtopic";
 
     // Setup holding vars for results from callback
-    private boolean sawCallback = false;
+    private boolean sawCallback;
     private static Exception actualException = null;
     private static List<ConsumerRecord<ByteString, ByteString>> actualRecords = null;
     private static List<TopicPartitionOffset> actualOffsets = null;
```

Comment:
> We can also have another field using a resettable latch (i.e. a `Phaser` or a 2-party `CyclicBarrier` if the `KafkaConsumerThreadPoolExecutor` thread can be blocked for a bit) that tests await in order to avoid arbitrary sleeps.

It actually might be simpler to just have the latch mimic the lifecycle of the state holder - add a `volatile CountDownLatch` field, recreate the latch in `setUp()`, countdown the latch in the `ConsumerReadCallback` callback.

Reply:
I actually undid the changes in `sawCallback` and `setUp`, and used a CountDownLatch in the test. Changing anything here would force me to changes pretty much all the tests, which I would rather do separately, preferably in `master`.